numba/tests/test_parfors.py

```diff
@@ -3668,7 +3683,7 @@ def test_impl():
         self.check(test_impl,)
         cpfunc = self.compile_parallel(test_impl, ())
         diagnostics = cpfunc.metadata['parfor_diagnostics']
-        self.assert_diagnostics(diagnostics, hoisted_allocations=1)
+        self.assert_diagnostics(diagnostics, hoisted_allocations=0)
```

Comment:
Why is this now not hoisted? I can guess from the proposed changes to analysis why it's the case but should it be the case?

Reply:
Fixed, I hope.